presto-hive/src/main/java/com/facebook/presto/hive/s3/PrestoS3ConfigurationUpdater.java

```diff
@@ -124,6 +128,12 @@ public void updateConfiguration(Configuration config)
         if (sseKmsKeyId != null) {
             config.set(S3_SSE_KMS_KEY_ID, sseKmsKeyId);
         }
+        config.setBoolean(S3_WEB_IDENTITY_ENABLED, webIdentityEnabled);
+        if (webIdentityEnabled) {
+            if (webIdentityTokenFile != null) {
+                config.set(S3_WEB_IDENTITY_TOKEN_FILE, webIdentityTokenFile);
+            }
+        }
```

Comment:
As these 2 are dependent config, I think we can set both after the condition check?
```suggestion
if (webIdentityEnabled && webIdentityTokenFile != null) {
    config.setBoolean(S3_WEB_IDENTITY_ENABLED, webIdentityEnabled);
    config.set(S3_WEB_IDENTITY_TOKEN_FILE, webIdentityTokenFile);
}
```

Reply:
These two configurations are related but not strictly dependent. `S3_WEB_IDENTITY_ENABLED` determines if web identity authentication is used, but the token file is optional. Keeping them separate ensures flexibility, as web identity mode can be enabled without necessarily requiring a token file.

Reply:
Ok in that case we can modify it to something like -
```
if (webIdentityEnabled && webIdentityTokenFile != null) {
    config.set(S3_WEB_IDENTITY_TOKEN_FILE, webIdentityTokenFile);
}
```